src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,127 +1,103 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
-    /**
-     * Верните текущую дату в виде строки в зависимости от запроса.
-     *
-     * @param datePart Запрос на часть даты или всю дата целиком:
-     *                 - FULL - текущая дата целиком год, месяц, день (число месяца)
-     *                 в формате YYYY-MM-DD, возвращаемое значение по умолчанию;
-     *                 - YEAR - текущий год;
-     *                 - MONTH - название текущего месяца;
-     *                 - DAY - текущий день (число месяца);
-     *                 В любом другом случае бросить DateTimeException
-     **/
+    private static final ZoneOffset ZONE_OFFSET_UKRAINE = ZoneOffset.of("+02:00");
+    private static final DateTimeFormatter FORMATTER1 = DateTimeFormatter.ofPattern("yyyyMMdd");
+    private static final DateTimeFormatter FORMATTER2 = DateTimeFormatter
+            .ofPattern("d MMM yyyy", Locale.ENGLISH);
+    private static final DateTimeFormatter FORMATTER3 = DateTimeFormatter
+            .ofPattern("dd LLLL yyyy HH:mm", Locale.ENGLISH);
+    private static final DateTimeFormatter FORMATTER4 = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private LocalDate today = LocalDate.now();
+
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        switch (datePart) {
+            case FULL:
+                return today.format(FORMATTER4);
+            case YEAR:
+                return String.valueOf(today.getYear());
+            case MONTH:
+                return String.valueOf(today.getMonth());
+            case DAY:
+                return String.valueOf(today.getDayOfWeek());
+            default:
+                throw new DateTimeException("Wrong format");
+        }
     }
 
-    /**
-     * Верните Optional даты соответствующей дате в массиве.
-     *
-     * @param dateParams Дан массив данных состоящий из 3-х элементов, где:
-     *                   - 1-й элемент массива - год;
-     *                   - 2-й элемент массива - месяц;
-     *                   - 3-й элемент массива - день (число);
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        if (dateParams.length < 3) {
+            return Optional.empty();
+        }
```

Comment:
try to refactor without this check by catching more than one Exception

Reply:
done